src/EFCore.PG/Extensions/NpgsqlNetworkExtensions.cs

```diff
@@ -0,0 +1,1006 @@
+        /// </exception>
+        public static bool ContainsOrEqual([CanBeNull] this DbFunctions _, (IPAddress Address, int Subnet) cidr, (IPAddress Address, int Subnet) other) => throw new ClientEvaluationNotSupportedException();
+
+        /// <summary>
+        /// Determines whether an <see cref="IPAddress"/> contains or is contained by another <see cref="IPAddress"/>.
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="inet">The IP address to search.</param>
+        /// <param name="other">The IP address to locate.</param>
+        /// <returns>
+        /// True if the <see cref="IPAddress"/> contains or is contained by the other <see cref="IPAddress"/>; otherwise, false.
+        /// </returns>
+        /// <exception cref="ClientEvaluationNotSupportedException">
+        /// This method is only intended for use via SQL translation as part of an EF Core LINQ query.
+        /// </exception>
+        public static bool ContainsOrContainedBy([CanBeNull] this DbFunctions _, IPAddress inet, IPAddress other) => throw new ClientEvaluationNotSupportedException();
+
+        /// <summary>
+        /// Determines whether an (IPAddress Address, int Subnet) contains or is contained by another (IPAddress Address, int Subnet).
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="cidr">The cidr to search.</param>
+        /// <param name="other">The cidr to locate.</param>
+        /// <returns>
+        /// True if the (IPAddress Address, int Subnet) contains or is contained by the other (IPAddress Address, int Subnet); otherwise, false.
+        /// </returns>
+        /// <exception cref="ClientEvaluationNotSupportedException">
+        /// This method is only intended for use via SQL translation as part of an EF Core LINQ query.
+        /// </exception>
+        public static bool ContainsOrContainedBy([CanBeNull] this DbFunctions _, (IPAddress Address, int Subnet) cidr, (IPAddress Address, int Subnet) other) => throw new ClientEvaluationNotSupportedException();
+
+        #endregion
+
+        #region BitwiseOperators
+
+        /// <summary>
+        /// Computes the bitwise NOT operation on an <see cref="IPAddress"/>.
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="inet">The inet to negate.</param>
+        /// <returns>
+        /// The result of the bitwise NOT operation.
+        /// </returns>
+        /// <exception cref="ClientEvaluationNotSupportedException">
+        /// This method is only intended for use via SQL translation as part of an EF Core LINQ query.
+        /// </exception>
+        public static IPAddress Not([CanBeNull] this DbFunctions _, IPAddress inet) => throw new ClientEvaluationNotSupportedException();
```

Comment:
I propose to rename to `BitwiseNot` to make things more clear (same for the other bit operations below), a "not" operation on an IP address is far from being an everyday thing. This is also how it is documented in the PostgreSQL docs.